Program.cs

```diff
@@ -32,13 +32,18 @@ public static void SaveObject(player[] thisObject)
         }
         public static void LoadObject(ref player[] thisObject)
         {
-            Stream sr;
-            BinaryFormatter bf = new BinaryFormatter();
-            sr = File.OpenRead("mydata.dat");
-            thisObject = (player[])bf.Deserialize(sr);
-            sr.Close();
-           
+            if (File.Exists("mydata.dat"))
```

Comment:
you weren't checking to see if mydata.dat existed or not (it didn't on my system, so I had to add a File.Exists check, and when the file doesn't exist, create a zero-length array of player objects so that the rest of your data storage logic can work.